### DISCUSSION TITLE
Random graphs batch generator

### DISCUSSION THREAD

**Discussion by qtomcatq** [networkx/networkx#7713] [Q&A] [answered]
Hello! 

I have a problem that requires me to generate a batch of Erdős-Rényi graphs, e.g. I would like to replace this for loop:

```
graph_batch=1024
dim=10
p=0.1
graph_list=[]
for i in range(graph_batch):
    G=nx.erdos_renyi_graph(dim,p)
    graph_list.append(G)
```

Is there any possibility of that?

**Comment by boothby** [comment, marked as answer]
Why not use a list comprehension?

```python
graph_list = [nx.erdos_renyi_graph(dim, p) for _ in range(graph_batch)]
```
While it would certainly be possible to implement such a function in NetworkX, you haven't given a motivation for doing so.  If it's just to cut down on linecount, then the above should be suitable.
